dataplane/saiserver/ports.go

```diff
@@ -52,19 +66,51 @@ type port struct {
 	nextEth   int
 	portToEth map[uint64]string
 	opts      *dplaneopts.Options
+	config    *dplaneopts.PortConfig
 }
 
 // stub for testing
 var getInterface = net.InterfaceByName
 
 // CreatePort creates a new port, mapping the port to ethX, where X is assigned sequentially from 1 to n.
 // Note: If more ports are created than eth devices, no error is returned, but the OperStatus is set to NOT_PRESENT.
-func (port *port) CreatePort(ctx context.Context, _ *saipb.CreatePortRequest) (*saipb.CreatePortResponse, error) {
+func (port *port) CreatePort(ctx context.Context, req *saipb.CreatePortRequest) (*saipb.CreatePortResponse, error) {
 	id := port.mgr.NextID()
 
+	// By default, create port sequentially starting at eth1.
 	dev := fmt.Sprintf("eth%v", port.nextEth)
 	port.nextEth++
 
+	// If a port config is set, then use the hardware lanes to find the interface names.
+	if port.config != nil {
+		if len(req.HwLaneList) == 0 {
+			return nil, fmt.Errorf("port lanes are required got %v", req.HwLaneList)
+		}
+		var b strings.Builder
+		b.WriteString(fmt.Sprint(req.HwLaneList[0]))
+		for _, l := range req.HwLaneList[1:] {
+			b.WriteString(",")
+			b.WriteString(fmt.Sprint(l))
+		}
+		dev = ""
+		lanes := b.String()
+		for n, cfg := range port.config.Ports {
+			if cfg.Lanes == lanes {
+				dev = n
+			}
+		}
+		if dev == "" {
+			return nil, fmt.Errorf("could not find lanes %v in config", lanes)
+		}
+		if port.opts.PortMap != nil && len(port.opts.PortMap) != 0 {
+			if portMapPort := port.opts.PortMap[dev]; portMapPort == "" {
+				log.Warningf("port maped port doesn't for port %v", dev)
```

Comment:
Can you revise the message? port mapped port not found?

Reply:
done